### DISCUSSION TITLE
Handling duplicates ManyToMany relationship | Documentation update

### DISCUSSION THREAD

**Discussion by KonstantinosVazaios** [filamentphp/filament#9450] [General]
I was reading the Filament Documentation about Relation Managers and specifically the sections about attaching & detaching records for a ManyToMany relationship. I needed to somehow allow the duplicate record attaches. I followed the documentation's steps in order to have that working. 

1. I defined the extra pivot columns within the withPivot() method
2. I added a primary id to the pivot table and also defined it in the withPivot() method
3. and finally I added the $allowsDuplicates property to the relation manager

However when I was to attach a record for the second time, the select input would not load the record again. I did a little research in the Filament classes and I noticed that inside the hasRecords trait there is a allowDuplicates() method. 
I added this method at the returned $table object inside the table() method of the RelationManager class and I got it working.

In a nutshell, the $allowsDuplicates property to the relation manager does not solve the problem. We need to add allowDuplicates() method in the $table object. 

We may need to update the Documentation

**Comment by zaidpirwani** [comment]
I followed the docs and it didnt allow me to add duplicates,
my pivot has id column and relations have the id mentioned in withPivot method.

Thanks to you, I was able to fix it

![image](https://github.com/filamentphp/filament/assets/1651976/0cf539aa-d609-4999-8978-287c868588ef)

**Reply by ayemin-tun** [reply]
Hello, Firstly Thanks for answering, this solution is work for my problem but I has some issues when I use allowDuplicates the attach data is allow duplicate but in bulk action  DetachBulkAction is not work for me. When I remove allowDuplicates the detachBulkaction is work. May I know your code has this problem.
